src/Compilers/CSharp/Test/Symbol/Compilation/CompilationAPITests.cs

```diff
@@ -45,6 +45,26 @@ public void PublicSignWithRelativeKeyPath()
             );
         }
 
+        [Fact]
+        [WorkItem(11497, "https://github.com/dotnet/roslyn/issues/11497")]
+        public void PublicSignWithEmptyKeyPath()
+        {
+            CreateStandardCompilation("", options: TestOptions.ReleaseDll.WithPublicSign(true).WithCryptoKeyFile("")).VerifyDiagnostics(
+                // error CS8102: Public signing was specified and requires a public key, but no public key was specified.
+                Diagnostic(ErrorCode.ERR_PublicSignButNoKey).WithLocation(1, 1));
+        }
+
+        [Fact]
+        [WorkItem(11497, "https://github.com/dotnet/roslyn/issues/11497")]
+        public void PublicSignWithEmptyKeyPath2()
+        {
+            CreateStandardCompilation("", options: TestOptions.ReleaseDll.WithPublicSign(true).WithCryptoKeyFile("\"\"")).VerifyDiagnostics(
+                // error CS8106: Option 'CryptoKeyFile' must be an absolute path.
```

Comment:
This error seems redundant (and also in VB). Is it easy to remove the cascade?

Reply:
We can. But that requires a minor API change (`AssemblyIdentity.HasPublicKey`), or considering empty strings as existing paths. I'm not sure if it was worth it to make that change.

---
In reply to: [156213571](https://github.com/dotnet/roslyn/pull/23646#discussion_r156213571) [](ancestors = 156213571)

Reply:
Ok. Makes sense to leave then.